src/DataFrames.jl

```diff
@@ -128,4 +128,18 @@ for (dir, filename) in [
     include(joinpath(dir, filename))
 end
 
+# Import  "StringEncodings" if available and set up decoder
+_decoder=nothing
+ENCODING_PACKAGE_READY = false
+function __init__()
+    try Pkg.installed("StringEncodings")
```

Comment:
The point is that what you end up doing by putting it in `__init__` is automatically break all other package, not automatically manages the precompile cache.